src/features/views/components/MoveViewDialog.tsx

```diff
@@ -0,0 +1,266 @@
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Link,
+  List,
+  ListItem,
+  useMediaQuery,
+} from '@mui/material';
+import { Box } from '@mui/system';
+import { useTheme } from '@mui/styles';
+
+import { useNumericRouteParams } from 'core/hooks';
+import ZUIFuture from 'zui/ZUIFuture';
+import { ZetkinViewFolder } from './types';
+import useViewBrowserItems, {
+  ViewBrowserFolderItem,
+  ViewBrowserItem,
+  ViewBrowserViewItem,
+} from '../hooks/useViewBrowserItems';
+import useViewBrowserMutations from '../hooks/useViewBrowserMutations';
+import useViewTree from '../hooks/useViewTree';
+import { useMessages } from 'core/i18n';
+import messageIds from '../l10n/messageIds';
+import BrowserItemIcon from './ViewBrowser/BrowserItemIcon';
+import { ViewTreeData } from 'pages/api/views/tree';
+
+const folderById = (id: number | null, viewTree: ViewTreeData) => {
+  if (id == null) {
+    return null;
+  }
+  return viewTree.folders.find((f) => f.id == id) ?? null;
+};
+
+const getAllParentFolderIds = (
+  folderId: number | null,
+  viewTree: ViewTreeData
+) => {
+  let parentFolderIds: number[] = [];
+
+  while (folderId != null) {
+    parentFolderIds = [folderId, ...parentFolderIds];
+
+    const folder = folderById(folderId, viewTree);
+    folderId = folder?.parent?.id ?? null;
+  }
+  return parentFolderIds;
+};
+
+const MoveItemBreadcrumbs = ({
+  onClickFolder,
+  orgId,
+  viewedFolder,
+}: {
+  onClickFolder: (folderId: number | null) => void;
+  orgId: number;
+  viewedFolder: number | null;
+}) => {
+  const messages = useMessages(messageIds);
+  const viewTreeFuture = useViewTree(orgId);
+
+  return (
+    <ZUIFuture future={viewTreeFuture}>
+      {(viewTree) => {
+        const folders = getAllParentFolderIds(viewedFolder, viewTree)
+          .map((id) => folderById(id, viewTree))
+          .filter((folder): folder is ZetkinViewFolder => !!folder);
+
+        // Add the root folder to the beginning
+        const breadcrumbItems = [
+          {
+            id: null,
+            title: messages.browserLayout.title(),
+          },
+          ...folders,
+        ];
+        return (
+          <Breadcrumbs
+            aria-label="breadcrumb"
+            itemsAfterCollapse={2}
+            itemsBeforeCollapse={1}
+            maxItems={3}
+            separator={<NavigateNextIcon fontSize="small" />}
+          >
+            {breadcrumbItems.map(({ title, id }) =>
+              id == viewedFolder ? (
+                <Box key={id ?? 'root'}>{title}</Box>
+              ) : (
+                <Link
+                  key={id ?? 'root'}
+                  color="inherit"
+                  onClick={() => onClickFolder(id)}
+                  sx={{ cursor: 'pointer' }}
+                  underline="hover"
+                >
+                  {title}
+                </Link>
+              )
+            )}
+          </Breadcrumbs>
+        );
+      }}
+    </ZUIFuture>
+  );
+};
+
+export type MoveViewDialogProps = {
```

Comment:
I don't think this should be exported.

Reply:
Sure thing! I often stumble upon cases where accessing prop types are useful, but if this is not a common pattern in Zetkin, I'll remove it.